plugins/aws/resoto_plugin_aws/resource/sns.py

```diff
@@ -0,0 +1,267 @@
+from typing import ClassVar, Dict, List, Optional, Type
+from attrs import define, field
+from resoto_plugin_aws.aws_client import AwsClient
+from resoto_plugin_aws.resource.base import AwsApiSpec, AwsResource, GraphBuilder
+from resoto_plugin_aws.resource.iam import AwsIamRole
+from resoto_plugin_aws.resource.kms import AwsKmsKey
+from resoto_plugin_aws.utils import ToDict
+from resotolib.baseresources import EdgeType, ModelReference
+from resotolib.json_bender import F, Bender, S, bend
+from resotolib.types import Json
+
+
+@define(eq=False, slots=False)
+class AwsSnsTopic(AwsResource):
```

Comment:
I wonder if we can introduce base classes here?

Reply:
Interesting thought. Can't see right now what that would be and which other resources we might group together there? 🤔

Reply:
Probably worth a thought and discussion - maybe not as part of this PR.